src/core/src/preprocess/color_utils.cpp

```diff
@@ -20,8 +20,10 @@ std::unique_ptr<ColorFormatInfo> ColorFormatInfo::get(ColorFormat format) {
         res.reset(new ColorFormatInfoI420_ThreePlanes(format));
         break;
     case ColorFormat::RGB:
+        res.reset(new ColorFormatRGB(format));
+        break;
     case ColorFormat::BGR:
-        res.reset(new ColorFormatNHWC(format));
+        res.reset(new ColorFormatBGR(format));
         break;
```

Comment:
Do you need to extend this switch-case for GRAY?

Reply:
By default for GRAY case it will be `ColorFormatInfo` class, which default layout is empty (`{}`). If we want to assume GRAY is NHWC input data, so yes, need to extend logic with that case

Reply:
Done